lib/pure/os.nim

```diff
@@ -2787,23 +2787,23 @@ when not weirdTarget and (defined(freebsd) or defined(dragonfly)):
     const KERN_PROC_PATHNAME = 9
 
   proc getApplFreebsd(): string =
-    var pathLength = csize_t(MAX_PATH)
-    result = newString(pathLength)
+    var pathLength = csize_t(0)
     var req = [CTL_KERN.cint, KERN_PROC.cint, KERN_PROC_PATHNAME.cint, -1.cint]
-    while true:
-      let res = sysctl(addr req[0], 4, cast[pointer](addr result[0]),
-                       addr pathLength, nil, 0)
-      if res < 0:
-        let err = osLastError()
-        if err.int32 == ENOMEM:
-          result = newString(pathLength)
-        else:
-          result.setLen(0) # error!
-          break
-      else:
-        # trim the trailing null byte, as the result is a string not a cstring
-        result.setLen(pathLength-1)
-        break
+
+    # first call to get the required length
+    var res = sysctl(addr req[0], 4, nil, addr pathLength, nil, 0)
+
+    if res < 0:
```

Comment:
IMO this should raise instead of ignoring the error (and setting the global errno)
```nim
raiseOSError(osLastError(), "something")
```
note that `getApplHeuristic` is buggy (and not even fixable) so that's not a good enough workaround

Reply:
This approach is copied from other implementations - see for example:

- [getApplgetApplHaiku](https://github.com/nim-lang/Nim/blob/devel/lib/pure/os.nim#L2903)
- [getApplHeuristic](https://github.com/nim-lang/Nim/blob/devel/lib/pure/os.nim#L2873)
- [getApplOpenBsd](https://github.com/nim-lang/Nim/blob/devel/lib/pure/os.nim#L2859)
- [Windows implementation](https://github.com/nim-lang/Nim/blob/devel/lib/pure/os.nim#L2937)
- [Mac implementation](https://github.com/nim-lang/Nim/blob/devel/lib/pure/os.nim#L2950)

IMO, it's better to be consistent - either all should raise on error or none should.